### PR TITLE
feat: avm/res/network/ip-group

### LINKED ISSUE #506: [Module Proposal]: `avm/res/network/ip-group`
### Check for previous/existing GitHub issues/module proposals

- [X] I have checked for previous/existing GitHub issues/module proposals.

### Check this module doesn't already exist in the module indexes

- [X] I have checked for that this module doesn't already exist in the module indexes; or I'm proposing the module to be migrated from CARML/TFVM.

### Bicep or Terraform?

Bicep

### Module Classification?

Resource Module

### Module Name

avm/res/network/ip-group

### Module Details

Migration from CARML
https://github.com/ahmadabdalla/bicep-registry-modules/tree/users/ahmad/4415_ipGroups

### Do you want to be the owner of this module?

Yes

### Module Owner's GitHub Username (handle)

ahmadabdalla

### (Optional) Secondary Module Owner's GitHub Username (handle)

_No response_

### LINKED ISSUE #4414: Align `network/ip-group` to AVM specs

### PR DESCRIPTION
**Pull request by ahmadabdalla**
### Pipeline

[avm.res.network.ip-group](https://github.com/ahmadabdalla/bicep-registry-modules/actions/workflows/avm.res.network.ip-group.yml)

### Teams

Closes https://github.com/Azure/Azure-Verified-Modules/issues/506
Closes https://github.com/Azure/ResourceModules/issues/4414

image

image

## Adding a new module

- [ ] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.

## Updating an existing module

- [ ] This is a bug fix:
  - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
  - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
- [ ] I have run `brm validate` locally to verify the module files.
- [ ] I have run deployment tests locally to ensure the module is deployable.
- [ ] I have read the [Updating an existing module](https://github.com/Azure/bicep-registry-modules/blob/main/CONTRIBUTING.md#updating-an-existing-module) section in the contributing guide and updated the `version.json` file properly:
  - [ ] The PR contains backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`.
  - [ ] The PR contains backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] The PR contains breaking changes, and I have bumped the MAJOR version in `version.json`.
- [ ] I have updated the examples in README with the latest module version number.

### CODE REVIEW

File: avm/res/network/ip-group/tests/e2e/waf-aligned/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+  params: {
+    name: '${namePrefix}${serviceShort}001'
+    location: resourceLocation
+    ipAddresses: [
+      '10.0.0.1'
+      '10.0.0.2'
+    ]
+    lock: {
```
Comment (eriqua) [inline avm/res/network/ip-group/tests/e2e/waf-aligned/main.test.bicep:49]: Can be removed for waf test

**Review by AlexanderSehr — DISMISSED** [review]
Good from my side. With @ChrisSidebotham we did not have a strong stance on the lock parameter @eriqua, so I will neither resolve nor object.

File: avm/res/network/ip-group/tests/e2e/waf-aligned/main.test.bicep (hunk 1 above)
Comment (ahmadabdalla) [reply avm/res/network/ip-group/tests/e2e/waf-aligned/main.test.bicep:49]: @eriqua  can this be resolved as per the comments from the team?

Comment (eriqua) [reply avm/res/network/ip-group/tests/e2e/waf-aligned/main.test.bicep:49]: As mentioned in another PR, we should remove locks from waf aligned tests, and better document waf aligned requirement in avm

Comment (ahmadabdalla) [reply avm/res/network/ip-group/tests/e2e/waf-aligned/main.test.bicep]: @eriqua removed from test as requested

File: .github/workflows/avm.res.network.ip-group.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      workflowInput: ${{ steps.get-workflow-param.outputs.workflowInput }}
+      moduleTestFilePaths: ${{ steps.get-module-test-file-paths.outputs.moduleTestFilePaths }}
+      psRuleModuleTestFilePaths: ${{ steps.get-module-test-file-paths.outputs.psRuleModuleTestFilePaths }}
+      modulePath: "${{ env.modulePath }}"
+
+  ##############################
+  #   Call reusable workflow   #
+  ##############################
+  call-workflow-passing-data:
+    name: "Run"
```
Comment (AlexanderSehr) [inline .github/workflows/avm.res.network.ip-group.yml:74]:
```suggestion
    name: "Run"
    permissions:
        id-token: write # For OIDC
        contents: write # For release tags
```

Required to account for latest GH updates. Without, the pipeline won't work in the `Azure` org.

**Review by AlexanderSehr — DISMISSED** [review]

File: .github/workflows/avm.res.network.ip-group.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        contents: write # For release tags
+    needs:
+      - job_initialize_pipeline
+    uses: ./.github/workflows/avm.template.module.yml
+    with:
+      workflowInput: "${{ needs.job_initialize_pipeline.outputs.workflowInput }}"
+      moduleTestFilePaths: "${{ needs.job_initialize_pipeline.outputs.moduleTestFilePaths }}"
+      psRuleModuleTestFilePaths: "${{ needs.job_initialize_pipeline.outputs.psRuleModuleTestFilePaths }}"
+      modulePath: "${{ needs.job_initialize_pipeline.outputs.modulePath}}"
+    secrets: inherit
```
Comment (AlexanderSehr) [inline .github/workflows/avm.res.network.ip-group.yml:85]:
```suggestion
    secrets: inherit

```

Prettier is failing - and I could imagine it is due to the line ending. Unfortunately it doesn't say, so let's just try.

**Review by AlexanderSehr — APPROVED** [review]

**Pull request merged**